src/plugins/intel_gpu/include/intel_gpu/runtime/memory.hpp

```diff
@@ -48,8 +48,8 @@ struct memory {
     virtual ~memory() = default;
     virtual void* lock(const stream& stream, mem_lock_type type = mem_lock_type::read_write) = 0;
     virtual void unlock(const stream& stream) = 0;
-    virtual event::ptr fill(stream& stream, unsigned char pattern, bool blocking = true) = 0;
-    virtual event::ptr fill(stream& stream, bool blocking = true) = 0;
+    virtual event::ptr fill(stream& stream, unsigned char pattern, const std::vector<event::ptr>& dep_events = {}, bool blocking = true) = 0;
+    virtual event::ptr fill(stream& stream, const std::vector<event::ptr>& dep_events = {}, bool blocking = true) = 0;
```

Comment:
Isn't it better to add the new arg as the last one?

Reply:
I just wanted to highlight its importance by placing it a bit earlier in the argument list, while still keeping it optional

Reply:
Got it your intention.